boot-qemu.sh

```diff
@@ -0,0 +1,199 @@
+#!/usr/bin/env bash
+
+# Root of the repo
+BASE=$(cd "$(dirname "$(readlink -f "${BASH_SOURCE[0]}")")" && pwd)
+
+
+# Prints an error message in bold red then exits
+function die() {
+    printf "\n\033[01;31m%s\033[0m\n" "${1}"
+    exit 1
+}
+
+
+# Check that a binary is found
+function checkbin() {
+    command -v "${1}" &>/dev/null || die "${1} could not be found, please install it!"
+}
+
+
+# Parse inputs to the script
+function parse_parameters() {
+    while (( ${#} )); do
+        case ${1} in
+            -a|--arch|--architecture)
+                shift
+                case ${1} in
+                    arm32_v5|arm32_v6|arm32_v7|arm64|mips|mipsel|ppc32|ppc64|ppc64le|x86_64) ARCH=${1} ;;
+                    *) die "Invalid --arch value '${1}'" ;;
+                esac ;;
+
+            -d|--debug)
+                set -x ;;
+
+            -h|--help)
+                echo
+                cat "${BASE}"/boot-qemu-help.txt
+                echo
+                exit 0 ;;
+
+            -i|--interactive|--shell)
+                INTERACTIVE=true ;;
+
+            -k|--kbuild-folder)
+                shift && KBUILD_DIR=${1} ;;
+
+            -t|--timeout)
+                shift && TIMEOUT=${1} ;;
+
+            *)
+                die "Invalid parameter '${1}'" ;;
+        esac
+        shift
+    done
+}
+
+
+# Sanity check parameters and required tools
+function sanity_check() {
+    # Kernel build folder and architecture are required paramters
+    # We are not interactive by default and the default timeout is 3m
+    : "${ARCH:?}" "${KBUILD_DIR:?}" "${INTERACTIVE:=false}" "${TIMEOUT:=3m}"
+
+    # KBUILD_DIR could be a relative path; turn it into an absolute one with readlink
+    KBUILD_DIR=$(readlink -f "${KBUILD_DIR}")
+
+    # Let the user know if the kernel build folder does not exist
+    [[ -d ${KBUILD_DIR} ]] || die "${KBUILD_DIR} does not exist!"
+
+    # Make sure zstd is install
+    checkbin zstd
+}
+
+
+# Decompress rootfs images
+function decomp_rootfs() {
+    [[ ${ARCH} =~ arm32 ]] && ARCH_RTFS_DIR=arm
+
+    IMAGES_DIR=${BASE}/images/${ARCH_RTFS_DIR:-${ARCH}}
+    ROOTFS=${IMAGES_DIR}/rootfs.cpio
+
+    rm -rf "${ROOTFS}"
+    zstd -d "${ROOTFS}".zst -o "${ROOTFS}"
+}
+
+
+# Boot QEMU
+function setup_qemu_args() {
+    if ${INTERACTIVE}; then
+        RDINIT=" rdinit=/bin/sh"
+        APPEND_RDINIT=( -append "${RDINIT}" )
+    fi
+
+    case ${ARCH} in
```

Comment:
Maybe create a small individual config for each ARCH value that can then be sourced?

Reply:
I would rather the values live here so that it is all visible for someone reading it. It makes the script a bit more verbose but it is better than having to veer off to some other file to see what values get set.

Reply:
I was thinking of how trivial would it be to add new arch in this case by just creating a config file. So if you want to share it, you would only share your config. I'm okay with current setup.

Reply:
Ack, thank you for the input. Maybe something to consider as we continue to grow. Would not necessarily be a bad idea to do at some point given that would make it easy for people to add their own values to this script. Will have to see how much we grow.